golang/pkg/builder/container/utils.go

```diff
@@ -176,6 +192,33 @@ func removeContainer(ctx context.Context, containerName string) error {
 	return nil
 }
 
+// Check the existence of a container, then return it, matches only one
+func getContainer(ctx context.Context, containerName string) (types.Container, error) {
```

Comment:
Can't we use the `GetContainersByName` from dagent utils?

Reply:
Not really since it's filter is not adequate sadly.

Reply:
```
	containers, err := cli.ContainerList(ctx, types.ContainerListOptions{
		All:     true,
		Filters: filters.NewArgs(filters.KeyValuePair{Key: "name", Value: fmt.Sprintf("^/?%s-", name)}),
	})
```
name filter could return more than one container, and it filters by a prefix judging by looks

Reply:
I would prefer to make smarter the first function and reuse it since this is a code duplication of the function doing the same thing, what do you think @nandor-magyar?

Reply:
Yes we could make it happen to avoid code duplication, the function needs  a boolean to toggle exact matching, in such case would prepend/append the regex parts if missing.
There an other thing troubling me here, that is builder using dagent code, not vice-versa -- that would be the correct way. All the docker functions we use could go into an isolated docker package since both of the use is aggressively.

Reply:
An approved way for me.

Reply:
We talked about it with @nandor-magyar , this is a much needed change, however it's a different scope, and will be implemented in different PR too.

Reply:
I don't like this approach, we'll forget it. If @nandor-magyar or you create a ticket for this, I can approve it.

Reply:
Or do it instantly after this PR 😄

Reply:
Made a ticket, will do it later this week. (I wanna finish my current task first)

Reply:
Thanks 🙏🏻